### PR TITLE
[Component with its own class name as service type] LPS-194841 Refactor Interpreter component

### PR DESCRIPTION
**Pull request by Valmir-unicap**
Related: https://liferay.atlassian.net/browse/LPS-194841

### CODE REVIEW

**Comment by Valmir-unicap** [comment]
ci:test:startup

**Comment by Valmir-unicap** [comment]
ci:test:lima

**Comment by Valmir-unicap** [comment]
ci:test:search

File: modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/interpreter/AssetRendererSearchResultInterpreter.java [hunk 1]
```diff
@@ -36,7 +36,7 @@
 /**
  * @author Wade Cao
  */
-@Component(service = AssetRendererSearchResultInterpreter.class)
+@Component(service = SearchResultInterpreter.class)
```
Comment (rafaprax) [inline modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/interpreter/AssetRendererSearchResultInterpreter.java:39]: Please inline this class into the SearchResultInterpreterProviderImpl.java

File: modules/apps/adaptive-media/adaptive-media-image-impl/src/main/java/com/liferay/adaptive/media/image/internal/handler/AMImageRequestHandler.java [hunk 2]
```diff
@@ -69,6 +74,89 @@ public AdaptiveMedia<AMProcessor<FileVersion>> handleRequest(
...
+			}
+
+			return Long.valueOf(matcher.group(2));
+		}
+
+		private static final Pattern _pattern = Pattern.compile(
+			"/image/(\\d+)(?:/(\\d+))?/([^/]+)/(?:[^/]+)");
+
+		@Reference
+		private AMImageConfigurationHelper _amImageConfigurationHelper;
```
Comment (rafaprax) [inline modules/apps/adaptive-media/adaptive-media-image-impl/src/main/java/com/liferay/adaptive/media/image/internal/handler/AMImageRequestHandler.java:153]: please move this to the outer class

**Review by rafaprax — CHANGES_REQUESTED** [review]

**Pull request closed without merging**